Mathlib/Set.lean

```diff
@@ -151,4 +154,29 @@ instance : LawfulFunctor Set where
      λ ⟨b, ⟨⟨a, ⟨h₁, h₂⟩⟩, h₃⟩⟩ => ⟨a, ⟨h₁, show h (g a) = c from h₂ ▸ h₃⟩⟩⟩
   map_const := rfl
 
+syntax (priority := high) "{ " term,* " }" : term
```

Comment:
```suggestion
syntax "{ " term,+ " }" : term
```
Otherwise this breaks the `{}` notation used for empty hashmaps, etc.

(I would also prefer no spaces inside the braces, but I realize that this is a subjective question.)

Reply:
Another thing we should do eventually is to add typeclasses for singleton and insert, but I'd still leave the built-in {} notation alone.

Reply:
> (I would also prefer no spaces inside the braces, but I realize that this is a subjective question.)

For this, I was following what you suggested at https://github.com/leanprover-community/mathlib4/pull/8#discussion_r638568899